### ISSUE TITLE
Showing the actual value of complex type parameter with NUnit in the reports

### ISSUE THREAD

**Issue by kuskmen** [nunit/docs#244]
Hi guys, 
first forgive me if this is not the right place to ask questions, but to me it is most suitable.

I remember that I've seen somewhere in the web about showing actual value of complex type parameter. Bear with me while I explain what I mean.

With this test:

    [Test]
    public void Post_StakeToDifferetnSelectionsViaBetsBetIdEndpoint_ReturnsCorrectData(
       [ValueSource(typeof(JsonContextCache), nameof(JsonContextCache.RandomThreeUniqueSelecionIds))] string[] selectionIds,
       [ValueSource(typeof(TestData), nameof(TestData.OddStyles))] OddStyle oddStyle, 
       [Values(true, false)] bool logged)

My report is generated as following: 

`Post_StakeToDifferetnSelectionsViaBetsBetIdEndpoint_ReturnsCorrectData(System.String[],European,True)`

What I want to achieve however is: 
`Post_StakeToDifferetnSelectionsViaBetsBetIdEndpoint_ReturnsCorrectData(["1","2","3"],European,True)`

Is it possible using nUnit or I am mistaken that I've seeing this in the web?

Thanks,
kuskmen

**Comment by CharliePoole** [comment]
To the extent this is something you want to be able to find out about in the docs, this is the right place for the issue. If you were asking us to change how NUnit names tests, then the right place would be under the NUnit framework repository at https://github.com/nunit/nunit

You can change the naming of __individual__ test cases in NUnit in certain circumstances:
1. When using the `[TestCase]` attribute, by specifying the `SetName` property.
2. When using the `[TestCaseSource]` attribute, provided the arguments are given by use of `TestCaseData` instances. Again, the property to use is `SetName`.

Two drawbacks to the above:
1. You have to do it one test at a time.
2. It isn't available at all when specifying arguments  using `[Values]` or `[ValueSource]`.

The general solution that NUnit has always historically advised is to override `ToString` in the types you are using as arguments. Obviously, when the type is `string[]` that's not available to you as an option.

Bottom line, there is no way to do this that I can see without a change to NUnit. Such a change has already been requested and is documented as issue nunit/nunit#1382. You're welcome to add comments there to encourage raising the priority of the issue.

I'm leaving this issue open and I think the @nunit/framework-team  should take a look at whether what I've explained above needs to be added to the documentation and, if so, where.

**Comment by MSK61** [comment]
A relevant issue in nunit3-vs-adapter, https://github.com/nunit/nunit3-vs-adapter/issues/78, had some comments that might be helpful in addressing this issue. In general I think solving this issue will definitely solve that other one.

**Comment by CharliePoole** [comment]
There are special issues when dealing with the Test Explorer, which would be mitigated if we implemented nunit/nunit#1382 but would still exist since it only resolves the problem for arrays but not for other complex types. That's due to the fact that NUnit tolerates duplicated test names, whereas Test Explorer doesn't. For the general case (using any runner, not just the NUnit 3 adapter under Test Explorer) nunit/nunit#1382 is the one to watch.
